### PR TITLE
Sample wallet GUI binaries (bundled with client-rpc)   (cro-402)

### PR DESCRIPTION
**Pull request by leejw51crypto**

### CODE REVIEW

File: spawn.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+if (require("electron-squirrel-startup")) {
+  app.quit();
+}
+
+function run_program() {
+  (async () => {
+    const { stdout } = await execa("/usr/local/bin/client-rpc", [
+      "--network-id",
+      "ab"
```
Comment (tomtau) [inline spawn.md]: Could these arguments be passed in externally from the main process?

**Comment by calvinlauco** [comment]
Should we provide the `app.js` instead of only the insturctions?

File: spawn.md (hunk 1 above)
Comment (calvinlauco) [inline spawn.md]: Better make program location, storage location and network id configurable via environment variable with a default value.

Comment (leejw51crypto) [reply spawn.md]: yes

File: spawn.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    ]);
+  })();
+}
+
+let mainWindow;
+
+const createWindow = () => {
+  mainWindow = new BrowserWindow({
+    width: 800,
+    height: 600
```
Comment (calvinlauco) [inline spawn.md]: Will have to do some test to see if the wallet works good on 800*600.

File: spawn.md (hunk 1 above)
Comment (leejw51crypto) [reply spawn.md]: yes
also for macosx, we should give absolute path to execa

File: spawn.md [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+```
+const { app, BrowserWindow } = require("electron");
+const { exec } = require("child_process");
+const execa = require("execa");
+
+if (require("electron-squirrel-startup")) {
+  app.quit();
+}
+
+function run_program() {
```
[the remark this answers is not in the document]
Comment (leejw51crypto) [reply spawn.md]: it's same.
latter one is simpler.

File: spawn.md (hunk 2 above)
Comment (leejw51crypto) [reply spawn.md]: what's default dimension?

File: spawn.md (hunk 1 above)
Comment (tomtau) [reply spawn.md]: And also the Tendermint RPC (HTTP + WS) parameters should be configurable, as one may not run the full node on its machine or configure it on different ports.
One could possibly bundle tendermint binary as well and launch `tendermint lite` as a proxy, but we had some issues with it before

**Comment by tomtau** [comment]
> Should we provide the `app.js` instead of only the instructions?

@leejw51crypto I think it'll make sense to include the file directly in the repository

**Comment by leejw51crypto** [comment]
ok , i will include it.
i will modify with above feedback.

**Comment by leejw51crypto** [comment]
current progress: testing with packaging manager
launching client-rpc part is ready

**Comment by leejw51crypto** [comment]
now building binaries for 3 oses

**Comment by leejw51crypto** [comment]
how to packaging)
yarn package  -> all

for each os)
yarn package-mac 
yarn package-linux
yarn package-win

File: main.js [hunk 4]
```diff
@@ -7,7 +7,11 @@ let win;
 
 function run_program() {
   (async () => {
-    const { stdout } = await execa("client-rpc", ["--network-id", "ab"]);
+    try {
+      const { stdout } = await execa("client-rpc", ["--network-id", "ab"]);
```
Comment (tomtau) [inline main.js]: can `["--network-id", "ab"]` (+ connection strings) be provided externally by the user?

**Comment by leejw51crypto** [comment]
spawning client-rpc works now
working on runtime argument from electron
currently support two options
--network-type or --n 
--network-id  or --i

File: spawn.md (hunk 1 above)
Comment (leejw51crypto) [reply spawn.md]: i will add two options
--tendermint-url
--websocket-url

File: spawn.md (hunk 2 above)
Comment (leejw51crypto) [reply spawn.md]: changed to 1024x768

Comment (leejw51crypto) [reply spawn.md]: also fixed dimension is also good

File: main.js (hunk 4 above)
Comment (leejw51crypto) [reply main.js]: yes

File: main.js [hunk 5]
```diff
@@ -5,17 +5,42 @@ const execa = require("execa");
 
 let win;
 
-function run_program() {
+function run_program(network_type, network_id) {
```
Comment (tomtau) [inline main.js]: @calvinaco could it just take `run_program(args)` and then `await execa("client-rpc", args);` ?

Comment (leejw51crypto) [reply main.js]: possible, but need to catch exception

**Pull request merged**